providers/discord/src/airflow/providers/discord/hooks/discord_webhook.py

```diff
@@ -148,11 +173,100 @@ def execute(self) -> None:
+    (https://discordapp.com/developers/docs/resources/webhook).
+
+    Each Discord webhook can be pre-configured to use a specific username and
+    avatar_url. You can override these defaults in this hook.
+
+    :param http_conn_id: Http connection ID with host as "https://discord.com/api/" and
+                         default webhook endpoint in the extra field in the form of
+                         {"webhook_endpoint": "webhooks/{webhook.id}/{webhook.token}"}
+    :param webhook_endpoint: Discord webhook endpoint in the form of
+                             "webhooks/{webhook.id}/{webhook.token}"
+    :param message: The message you want to send to your Discord channel
+                    (max 2000 characters)
+    :param username: Override the default username of the webhook
+    :param avatar_url: Override the default avatar of the webhook
+    :param tts: Is a text-to-speech message
+    :param proxy: Proxy to use to make the Discord webhook call
+    """
+
+    default_headers = {
+        "Content-Type": "application/json",
+    }
+    conn_name_attr = "http_conn_id"
+    default_conn_name = "discord_default"
+    conn_type = "discord"
+    hook_name = "Async Discord"
+
+    def __init__(
+        self,
+        *,
+        http_conn_id: str = "",
+        webhook_endpoint: str | None = None,
+        message: str = "",
+        username: str | None = None,
+        avatar_url: str | None = None,
+        tts: bool = False,
+        proxy: str | None = None,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.http_conn_id = http_conn_id
+        self.webhook_endpoint = webhook_endpoint
+        self.message = message
+        self.username = username
+        self.avatar_url = avatar_url
+        self.tts = tts
+        self.proxy = proxy
+        self.handler = DiscordCommonHandler()
+
+    async def _get_webhook_endpoint(self) -> str:
+        """
+        Return the default webhook endpoint or override if a webhook_endpoint is manually supplied.
+
+        :param http_conn_id: The provided connection ID
+        :param webhook_endpoint: The manually provided webhook endpoint
+        :return: Webhook endpoint (str) to use
+        """
+        conn = None
+        if not self.webhook_endpoint and self.http_conn_id:
+            conn = await get_async_connection(self.http_conn_id)
+        return self.handler.get_webhook_endpoint(conn, self.webhook_endpoint)
+
+    async def execute(self) -> None:
+        """Execute the Discord webhook call."""
+        webhook_endpoint = await self._get_webhook_endpoint()
+        discord_payload = self.handler.build_discord_payload(
+            tts=self.tts, message=self.message, username=self.username, avatar_url=self.avatar_url
+        )
+
+        session_kwargs: dict[str, Any] = {}
+        if self.proxy:
+            session_kwargs["proxy"] = self.proxy
+
+        async with aiohttp.ClientSession(**session_kwargs) as session:
```

Comment:
It seems we can set the `proxy` kwarg (and `proxies` for sync one) directly.

```suggestion
        async with aiohttp.ClientSession(proxy=self.proxy) as session:
```

Reply:
looks like right. After @dondaum confirmed, I think we can merge this one

Reply:
Thanks for the comment. Yes you are right. I checked `aiohttp.ClientSession` and `proxy` has `proxy: Optional[StrOrURL] = None`. I changed it.